app.test.js

```diff
@@ -0,0 +1,37 @@
+const app = require('./app')
+const request = require('supertest')
+const db = require('./db/connection')
+const seed = require('./db/seeds/seed')
+const data = require('./db/data/test-data')
+
+beforeEach(() => {
+    return seed(data)
+   })
+   
+   afterAll(() => {
+       return db.end()
+   })
+
+
+describe("GET /api/topics", () => {
+    test("returns status of code 200 when a request is made to this endpoint", () => {
+        return request(app).get("/api/topics").expect(200);
+    })
+    test("Returns topics with the correct format", () => {
+        return request(app)
+        .get('/api/topics')
+        .then(({body}) => {
+          const topics = body
+          expect(topics).toHaveLength(3)
+          topics.forEach((topic) => {
+            expect(typeof topic.slug).toBe("string")
+            expect(typeof topic.description).toBe("string")
+          })
+        }) 
+    })
+    test("handles API endpoint not found (404)", () => {
+        return request(app)
+          .get("/api/nonexistentendpoint")
+          .expect(404)
+    }) 
```

Comment:
This is a good test but for clarity it's worth putting this in it's own describe block as it's not really part of the `/api/topics` endpoint.

Reply:
Ok thanks for the feedback. I included it outside as you said.